src/snackbar/snackbar.js

```diff
@@ -0,0 +1,161 @@
+  this.textElement_ = document.createElement('div');
+  this.snackbarElement_.classList.add(this.cssClasses.snackbar);
+  this.textElement_.classList.add(this.cssClasses.message);
+  this.snackbarElement_.appendChild(this.textElement_);
+  this.snackbarElement_.setAttribute('aria-hidden', true);
+
+  if (this.actionHandler_) {
+    this.actionElement_ = document.createElement('button');
+    this.actionElement_.type = 'button';
+    this.actionElement_.classList.add(this.cssClasses.action);
+    this.actionElement_.textContent = this.actionText_;
+    this.snackbarElement_.appendChild(this.actionElement_);
+    this.actionElement_.addEventListener('click', this.actionHandler_);
+  }
+
+  this.element_.appendChild(this.snackbarElement_);
+  this.textElement_.textContent = this.message_;
+  this.snackbarElement_.classList.add(this.cssClasses.activeSnackbar);
+  this.snackbarElement_.setAttribute('aria-hidden', false);
+  setTimeout(this.cleanup.bind(this), this.timeout_);
+
+};
+
+MaterialSnackbar.prototype.removeSnackbar = function() {
+  'use strict';
+  if (this.actionElement_ && this.actionElement_.parentNode) {
+    this.actionElement_.parentNode.removeChild(this.actionElement_);
+  }
+  this.textElement_.parentNode.removeChild(this.textElement_);
+  this.snackbarElement_.parentNode.removeChild(this.snackbarElement_);
+};
+
+MaterialSnackbar.prototype.showSnackbar = function(data) {
+  'use strict';
+  if (data === undefined) {
+    throw new Error(
+      'Please provide a data object with at least a message to display.');
+  }
+  if (data.message === undefined) {
+    throw new Error('Please provide a message to be displayed.');
+  }
+  if (data.actionHandler && !data.actionText) {
+    throw new Error('Please provide action text with the handler.');
+  }
+  if (this.active) {
+    this.queuedNotifications.push(data);
+  } else {
+    this.active = true;
+    this.message_ = data.message;
+    if (data.timeout) {
+      this.timeout_ = data.timeout;
+    } else {
+      this.timeout_ = this.defaults.timeout;
+    }
+    if (data.actionHandler) {
+      this.actionHandler_ = data.actionHandler;
+    }
+    if (data.actionText) {
+      this.actionText_ = data.actionText;
+    }
+    this.createSnackbar();
+  }
+};
+
+MaterialSnackbar.prototype.checkQueue = function() {
+  'use strict';
+  if (this.queuedNotifications.length > 0) {
+    this.showSnackbar(this.queuedNotifications.shift());
+  }
+};
+
+MaterialSnackbar.prototype.cleanup = function() {
+  'use strict';
+  this.snackbarElement_.classList.remove(this.cssClasses.activeSnackbar);
+  this.snackbarElement_.setAttribute('aria-hidden', true);
+  if (this.actionElement_) {
+    this.actionElement_.removeEventListener('click', this.actionHandler_);
+  }
+  this.setDefaults_();
+  this.active = false;
+  this.removeSnackbar();
+  this.checkQueue();
+};
+
+MaterialSnackbar.prototype.setDefaults_ = function() {
+  'use strict';
+  this.actionHandler_ = undefined;
```

Comment:
so you see people using "class" as a long lived object that people will call cleanup and init on multiple times?